File: backend/src/main/java/com/cduestc/openatom/activity/entity/Activity.java
package com.cduestc.openatom.activity.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("activities")
public class Activity {
    @TableId(type = IdType.ASSIGN_ID)
    private Long id;
    
    private String title;
    private String description;
    private String content;
    private String coverImage;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private String location;
    private Integer maxParticipants;
    private Integer currentParticipants;
    private Integer pointsReward;
    private Long organizerId;
    private String status; // upcoming, ongoing, completed, cancelled
    
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    
    @TableLogic
    private Boolean deleted;
}

File: frontend/src/types/activity.ts
export interface Activity {
  id: number
  title: string
  description: string
  content: string
  coverImage: string
  startTime: string
  endTime: string
  location: string
  maxParticipants: number
  currentParticipants: number
  pointsReward: number
  organizerId: number
  organizerName?: string
  status: string // upcoming, ongoing, completed, cancelled
  createdAt: string
  updatedAt: string
}

export interface ActivityQuery {
  page: number
  size: number
  status?: string
  keyword?: string
}

export interface ActivityRegistration {
  id: number
  activityId: number
  userId: number
  status: string // registered, attended, cancelled
  registeredAt: string
}
